### PR TITLE
Features/small improvements

### LINKED ISSUE #142: Numpy np.float128 bug in API template
### Describe the bug

This line :
https://github.com/OSS-Pole-Emploi/gabarit/blob/30137a556b0e5431f457c1a6046ac2c27525a339/gabarit/template_api/api_project/tests/test_schemas_utils.py#L26

sometimes makes the test crash on windows.

## Concerned template

  - [ ] NLP template
  - [ ] NUM template
  - [ ] VISION template
  - [x] API template
  - [ ] How templates are generated - Jinja

### Expected behavior

It should not crash !

### Actual behavior

It crashes !

### Additional context

Solution : use np.longdouble instead

### PR DESCRIPTION
**Pull request by gsolard**
## ✒️ Context

Closes issues #142 , #137 and #100

- What kind of change does this PR introduce ?

  - [x] Bugfix
  - [ ] Feature
  - [x] Refactoring
  - [ ] Other, please describe:

## 🧱 Description of Changes

- Fixes a bug when using np.float128
- Now properly uses NAME_VENV in the API template makefile
- Got rid of all the assert (except in tests) to be sure they don't disappear when compiling the code

  - [x] This is a change for the NLP template
  - [x] This is a change for the NUM template
  - [x] This is a change for the VISION template
  - [x] This is a change for the API template
  - [ ] This changes how templates are generated (i.e. a Jinja change)

## 🩺 Testing

- _Add bullet points summarizing your changes here_

  - [ ] This change does not need new tests
  - [x] Added/Updated unit tests
  - [ ] Added/Updated functionals tests (i.e. e2e)

## 🔗 References

- **Issue**: Closes  #142 , #137 and #100

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the GNU AFFERO GENERAL PUBLIC LICENSE.

### CODE REVIEW

File: gabarit/template_nlp/nlp_project/package_name-scripts/2_training.py [hunk 1]
```diff
@@ -141,12 +141,11 @@ def main(filename: str, x_col: Union[str, int], y_col: List[Union[str, int]], fi
...
-        try:
-            df_train[y_col] = df_train[y_col].astype(int)  # Need to cast OHE var into integers
-            for col in y_col:
-                assert sorted(df_train[col].unique()) == [0, 1]
-        except Exception:
-            raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
+        df_train[y_col] = df_train[y_col].astype(int)  # Need to cast OHE var into integers
+        for col in y_col:
+            if sorted(df_train[col].unique()) != [0, 1]:
+                raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
```
Comment (Eilwenn) [inline gabarit/template_nlp/nlp_project/package_name-scripts/2_training.py:147]: Should we specify the column name `col` in the value error ?

File: gabarit/template_nlp/nlp_project/package_name-scripts/3_predict.py [hunk 2]
```diff
@@ -116,12 +116,10 @@ def main(filename: str, x_col: Union[str, int], model_dir: str, y_col: Union[Lis
...
-            try:
-                y_true = df[y_col].astype(int)  # Need to cast OHE var into integers
-                for col in y_col:
-                    assert sorted(y_true[col].unique()) == [0, 1]
-            except Exception:
-                raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
+            y_true = df[y_col].astype(int)  # Need to cast OHE var into integers
+            for col in y_col:
+                if sorted(y_true[col].unique()) != [0, 1]:
+                    raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
```
Comment (Eilwenn) [inline gabarit/template_nlp/nlp_project/package_name-scripts/3_predict.py:122]: Should we specify the column name col in the value error ?

File: gabarit/template_nlp/nlp_project/package_name-scripts/utils/0_sweetviz_report.py [hunk 3]
```diff
@@ -135,8 +135,8 @@ def get_paths_and_names(dataset_paths: List[str], dataset_names: List[str]) -> T
         dataset_paths (list<str>): List of datasets filenames (actually paths relative to {{package_name}}-data)
         dataset_names (list<str>): List of names to use for each dataset. If no name provided, backup on files names without extension.
     Raises:
-        AssertionError: If any provided dataset does not exist
-        AssertionError: If a provided list of name does not the correct number of datasets
+        FileNotFoundError: If any provided dataset does not exist
+        ValueError: If a provided list of name does not the correct number of datasets
```
Comment (Eilwenn) [inline gabarit/template_nlp/nlp_project/package_name-scripts/utils/0_sweetviz_report.py:139]: If a provided list of names does not matches/equals the correct number of datasets

File: gabarit/template_nlp/nlp_project/package_name/models_training/models_tensorflow/utils_deep_keras.py [hunk 4]
```diff
@@ -323,7 +323,8 @@ def get_config(self) -> Any:
         return config
 
     def build(self, input_shape) -> None:
-        assert len(input_shape) == 3
+        if len(input_shape) != 3:
+            raise ValueError("input_shape should be 3")
```
Comment (Eilwenn) [inline gabarit/template_nlp/nlp_project/package_name/models_training/models_tensorflow/utils_deep_keras.py:327]: Should we add a "raises" section in the class docstring ? Or do we wait for a refacto of the experimental parts ?

File: gabarit/template_num/num_project/package_name-scripts/4_predict.py [hunk 5]
```diff
@@ -124,10 +124,11 @@ def main(filename: str, model_dir: str, y_col: Union[List[Union[str, int]], None
...
                 try:
                     y_true = df[y_col].astype(int)  # Need to cast OHE var into integers
-                    for col in y_col:
-                        assert sorted(y_true[col].unique()) == [0, 1]
-                except Exception:
+                except:
                     raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
+                for col in y_col:
+                    if sorted(y_true[col].unique()) != [0, 1]:
+                        raise ValueError("You provided several target columns, but at least one of them does not seem to be in a correct OHE format.")
```
Comment (Eilwenn) [inline gabarit/template_num/num_project/package_name-scripts/4_predict.py:131]: Should we specify the column name col in the value error ?

**Review by Eilwenn — CHANGES_REQUESTED** [review]

File: gabarit/template_nlp/nlp_project/package_name/models_training/models_tensorflow/utils_deep_keras.py (hunk 4 above)
Comment (gsolard) [reply gabarit/template_nlp/nlp_project/package_name/models_training/models_tensorflow/utils_deep_keras.py:327]: I think we should wait for the refactoring of the experimental parts

File: gabarit/template_num/num_project/package_name-tutorials/tutorial_exercices/verify.py [hunk 6]
```diff
@@ -113,26 +105,18 @@ def verify_exercice_5():
...
+    
     file_path = DATA_PATH / f"{DATASET_NAME}_preprocess_P2.csv"
-    try:
-        assert os.path.exists(file_path)
-    except AssertionError:
-        raise AssertionError(
-            "Did you run 1_preprocess_data.py with argument '-p preprocess_P2' ?"
-        )
+    if not file_path.exists():
+        raise AssertionError("Did you run 1_preprocess_data.py with argument '-p preprocess_P2' ?")
```
[the remark this answers is not in the document]
Comment (gsolard) [reply gabarit/template_num/num_project/package_name-tutorials/tutorial_exercices/verify.py]: I don't know why the test is different. Changed it

**Review by Eilwenn — APPROVED** [review]

**Pull request merged**